src/main/scala/zio/nio/channels/AsynchronousFileChannel.scala

```diff
@@ -1,30 +1,26 @@
 package zio.nio.channels
 
 import java.io.IOException
-import java.nio.channels.{ FileLock, AsynchronousFileChannel => JAsynchronousFileChannel }
+import java.nio.channels.{ AsynchronousFileChannel => JAsynchronousFileChannel, FileLock => JFileLock }
 import java.nio.file.attribute.FileAttribute
 import java.nio.file.{ OpenOption, Path }
 import java.util.concurrent.ExecutorService
 
+import zio.blocking.Blocking
 import zio.nio.{ Buffer, ByteBuffer }
-import zio.{ Chunk, IO, UIO }
+import zio.{ Chunk, IO, ZIO }
 
 import scala.collection.JavaConverters._
 
-class AsynchronousFileChannel(private val channel: JAsynchronousFileChannel) {
+class AsynchronousFileChannel(protected val channel: JAsynchronousFileChannel) extends Channel {
 
   import AsynchronousChannel.wrap
 
-  final def close: IO[Exception, Unit] =
-    IO.effect(channel.close()).refineToOrDie[Exception]
-
   final def force(metaData: Boolean): IO[IOException, Unit] =
     IO.effect(channel.force(metaData)).refineToOrDie[IOException]
 
-  final val lock: IO[Exception, FileLock] = wrap[FileLock](channel.lock((), _))
-
-  final def lock(position: Long, size: Long, shared: Boolean): IO[Exception, FileLock] =
-    wrap[FileLock](channel.lock(position, size, shared, (), _))
+  final def lock(position: Long = 0L, size: Long = Long.MaxValue, shared: Boolean = false): IO[Exception, FileLock] =
+    wrap[JFileLock](channel.lock(position, size, shared, (), _)).map(new FileLock(_))
 
   final private[nio] def readBuffer(dst: ByteBuffer, position: Long): IO[Exception, Int] =
```

Comment:
I was thinking about expose this to API. it would be much easier when you need to implement your own protocol.

Reply:
Yes, I was thinking the same thing, I think it can cut down copying in many cases. I think there's a few places where we want to make these public.